index.js

```diff
@@ -52,44 +55,64 @@ module.exports = {
       babelPlugins = addonOptions.babel.plugins = addonOptions.babel.plugins || [];
     }
 
-    // borrowed from ember-cli-htmlbars http://git.io/vJDrW
-    let projectConfig = this.projectConfig() || {};
-    let EmberENV = projectConfig.EmberENV || {};
-    let templateCompilerPath = this.templateCompilerPath();
-
-    // ensure we get a fresh templateCompilerModuleInstance per ember-addon
-    // instance NOTE: this is a quick hack, and will only work as long as
-    // templateCompilerPath is a single file bundle
-    //
-    // (╯°□°）╯︵ ɹǝqɯǝ
-    //
-    // we will also fix this in ember for future releases
-    delete require.cache[templateCompilerPath];
-
-    global.EmberENV = EmberENV;
-
-    let pluginInfo = this.astPlugins();
-    let Compiler = require(templateCompilerPath);
-    let templateCompilerFullPath = require.resolve(templateCompilerPath);
-    let templateCompilerCacheKey = fs.readFileSync(templateCompilerFullPath, { encoding: 'utf-8' });
-
-    pluginInfo.plugins.forEach(function(plugin) {
-      Compiler.registerPlugin('ast', plugin);
-    });
-
-    let precompile = Compiler.precompile;
-
-    precompile.baseDir = () => __dirname;
-    precompile.cacheKey = () => [templateCompilerCacheKey].concat(pluginInfo.cacheKeys).join('|');
-
-    delete require.cache[templateCompilerPath];
-    delete global.Ember;
-    delete global.EmberENV;
+    let pluginWrappers = this.parentRegistry.load('htmlbars-ast-plugin');
 
     // add the HTMLBarsInlinePrecompilePlugin to the list of plugins used by
     // the `ember-cli-babel` addon
     if (!this._registeredWithBabel) {
-      babelPlugins.push([HTMLBarsInlinePrecompilePlugin, { precompile  }]);
+      let templateCompilerPath = this.templateCompilerPath();
+      let parallelConfig = this.getParallelConfig(pluginWrappers);
+      if (this.canParallelize(pluginWrappers)) {
+        _logger.debug('using parallel API with broccoli-babel-transpiler');
+        babelPlugins.push({
+          _parallelBabel: {
+            requireFile: path.resolve(__dirname, 'require-from-worker'),
+            buildUsing: 'build',
+            params: {
+              templateCompilerPath,
+              parallelConfig
+            }
+          }
+        });
+      }
+      else {
+        _logger.debug('NOT using parallel API with broccoli-babel-transpiler');
```

Comment:
It would be nice if this function, and the one invoked while parallelized was the same function. I fear the risk of them diverging is high otherwise. And it would be nice to ensure the same tests could be used.